frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -276,9 +300,35 @@ export const tabsReducer = createReducer<DashboardState>(
     );
 
     builder.addCase(selectTab, (state, { payload: { tabId } }) => {
-      state.selectedTabId = tabId;
+      _selectTab({ state, tabId });
     });
 
+    builder.addCase(
+      moveDashCardToTab,
+      (state, { payload: { dashCardId, destTabId } }) => {
+        const { dashId } = getPrevDashAndTabs({ state });
+        if (dashId === null) {
+          throw Error(
+            `MOVE_DASHCARD_TO_TAB was dispatched but dashId (${dashId}) is null`,
+          );
+        }
+        const dashCard = state.dashcards[dashCardId];
+
+        const { row, col } = getPositionForNewDashCard(
+          // @ts-expect-error: "Type instantiation is excessively deep and possibly infinite."
+          // Unfortunate problem that comes with using rtk draft state.
+          getExistingDashCards(state, dashId, destTabId),
```

Comment:
@npretto just change the function definition

```
export function getExistingDashCards(
  dashboardState: Draft<DashboardState>,
```

I've verified locally that it works